File: python/problem167.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1
        while left < right:
            sum2num = numbers[left] + numbers[right]
            if sum2num == target:
                return [left + 1, right + 1]
            elif sum2num < target:
                left += 1
            else:
                right -= 1
        return []

File: Java/problem167.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int left = 0, right = numbers.length - 1;
        while (left < right) {
            int sum2num = numbers[left] + numbers[right];
            if (sum2num == target) {
                return new int[] {left + 1, right + 1};
            }
            else if (sum2num < target) {
                ++left;
            }
            else {
                --right;
            }
        }
        return null;
    }
}
